src/events/auth.py

```diff
@@ -67,11 +78,5 @@ def _on_auth_db_relation_removed(self, event) -> None:
 
     @compute_status
     def _on_auth_db_relation_departed(self, event) -> None:
-        """Handle the event when the authentication database relation is departed.
-
-        Until this point, the relation data is still there and hence the credentials
-        can be fetched for one last time in order to remove authentication database.
-        """
+        """Handle the event when the authentication database relation is departed."""
         self.logger.info("Authentication database relation departed")
-        auth = AuthenticationManager(self.context.auth_db)
-        auth.remove_auth_db()
```

Comment:
I had my suspicion that this was an issue in the housekeeping PR and made a note there. Turns out, it was indeed an issue and a blind spot in our test coverage: this event is emitted when we scale down kyuubi, meaning that we would remove the user table and thus break the charm.

Reply:
Great catch Alex!

Reply:
> I think this should probably be done in relation-broken hook instead

It's not the requirer responsibility to cleanup any resource on the provider side, right?